components/Chart.js

```diff
@@ -2,33 +2,39 @@ import React, { useState } from "react";
 import styled from "styled-components";
 import { BarChart, YAxis, XAxis, Bar } from "recharts";
 import useSWR from "swr";
+import { capitalizeFirstLetter } from "@/lib/utils";
 
 export default function BarChartPage() {
   const { data: transactionsList = [] } = useSWR("/api/transactions");
   const [selectedCategory, setSelectedCategory] = useState(null);
+  const [transactionType, setTransactionType] = useState("income");
```

Comment:
This is fine for now, in a bigger project we would use "enums" to prevent typos when distinguishing between transaction types. JS doesnt have enums, but we can use objects to simulate them: 
```js
const TRANSACTION_TYPES = {
  income: "income",
  expense: "expense"
}
```
Then you can use these object keys instead of strings in the state as well as in the transactions:

```js
const newTransaction = {
  ...
  type: TRANSACTION_TYPES.income
}

const [transactionType, setTransactionType] = useState(TRANSACTION_TYPES.income);
```
For your app this is not needed, but it is good to know this concept when working in bigger projects.